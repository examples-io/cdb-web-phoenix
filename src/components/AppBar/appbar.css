@media screen and (min-width: 0px) {
    .app-bar {
        border-top: 4px solid #ed0722;
        text-align: center;
        height: 50px;
        width: 100%;
        min-height: 11rem;
    }

    .app-bar-left {
        position: relative;
        float: left;
        margin: 3.6rem;
    }

    .app-bar-right {
        position: relative;
        float: right;
        margin: 3.6rem;
    }
}

@media screen and (min-width: 640px) {

}

@media screen and (min-width: 1200px) {

}