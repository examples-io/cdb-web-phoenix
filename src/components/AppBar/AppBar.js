import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './appbar.css'

/**
 * @description Header component - renders a header with optional left and right corner spaces for logos, etc.
 */
class AppBar extends Component {
    render() {
        return (
            <div className="app-bar">
                <div className="app-bar-left">
                    {this.props.headerLeft}
                </div>
                { this.props.children && <h1> {this.props.children} </h1> }
                <div className="app-bar-right">
                    {this.props.headerRight}
                </div>
            </div>
        );
    }
}

AppBar.propTypes = {
    /**
     * headerLeft - optional content to be displayed in the left corner of the header
     */
    headerLeft: PropTypes.node,
    /**
     * headerRight - optional content to be displayed in the right corner of the header
     */
    headerRight: PropTypes.node,
    /**
     * children - optional content to be displayed in the center of the header
     */
    children: PropTypes.node
};

export default AppBar;
