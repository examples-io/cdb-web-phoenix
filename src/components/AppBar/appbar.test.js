import React from 'react';
import renderer from 'react-test-renderer';

import AppBar from './AppBar';

describe('AppBar tests', () => {
    it ('renders Header correctly', () => {
        const tree = renderer
            .create(<AppBar headerLeft="left logo here" headerRight="right logo here" > Test Header </AppBar>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
