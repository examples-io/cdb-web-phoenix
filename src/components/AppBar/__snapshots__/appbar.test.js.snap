// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar tests renders Header correctly 1`] = `
<div
  className="app-bar"
>
  <div
    className="app-bar-left"
  >
    left logo here
  </div>
  <h1>
     
     Test Header 
     
  </h1>
  <div
    className="app-bar-right"
  >
    right logo here
  </div>
</div>
`;
